### PR TITLE
chore: refactor profile detail

### PR DESCRIPTION
**Pull request by andyogaga**
https://app.clickup.com/t/2kb4j0u

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
 import Home from "../views/Home.vue";
 import Profile from "../views/Profile.vue";
 import ProfileUpload from "../components/profile/profile-upload/ProfileUpload.vue";
-import ProfileDetail from "../components/profile/profile-detail/ProfileDetail.vue";
+import ProfileDetail from "@/views/profile/ProfileDetail.vue";
```
Comment (dzbo) [inline src/router/index.ts:5]: Let's name in similar way as upload on Idowu PR, so `ProfileDetail.vue` -> `Detail.vue`

File: src/views/profile/ProfileDetail.vue [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      error.value = err.toString();
+    });
+};
+
+const getProfileDataFromIPFS = (ipfsHash: string) => {
+  if (!ipfsHash) {
+    return false;
+  }
+
+  fetch("https://2eff.lukso.dev/ipfs/" + ipfsHash)
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:50]: Get from config

File: src/views/profile/ProfileDetail.vue [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  if (isAddress(addressOrHash)) {
+    getProfileDataFromERC725Cache(addressOrHash);
+  } else {
+    getProfileDataFromIPFS(addressOrHash);
+  }
+};
+
+const getProfileDataFromERC725Cache = (fetchedAddress: string) => {
+  fetchProfile(fetchedAddress)
+    .then((result) => {
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:33]: Use `async`/`await` instead of the callbacks

File: src/views/profile/ProfileDetail.vue [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import { isAddress } from "ethers/lib/utils";
+import useErc725 from "@/compositions/useErc725";
+import { useRoute, RouteLocationNormalizedLoaded } from "vue-router";
+
+const route = useRoute();
+
+const routeData = ref<RouteLocationNormalizedLoaded>(route);
+const dataSource = ref("");
+const loading = ref(true);
+const profileData = ref(null as any);
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:13]: Please avoid `any` when returned type is known, use `LSP3Profile` from `@lukso/lsp-factory.js` instead

```diff
- const profileData = ref(null as any);
+ const profileData = ref<{ LSP3Profile: LSP3Profile }>();
```

File: src/views/profile/ProfileDetail.vue [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        <th>Value</th>
+      </tr>
+      <tr>
+        <td>Name</td>
+        <td>{{ profileData?.LSP3Profile?.name }}</td>
+      </tr>
+
+      <tr>
+        <td>Description</td>
+        <td>{{ profileData?.LSP3Profile?.value }}</td>
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:92]: `profileData?.LSP3Profile?.description`?

File: src/views/profile/ProfileDetail.vue [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      </tr>
+
+      <tr>
+        <td>Links</td>
+        <td>
+          <ul class="list">
+            <li v-for="link in profileData?.LSP3Profile?.links" :key="link">
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:99]: I think index would be better as key, link is an object btw.

File: src/views/profile/ProfileDetail.vue [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+            />
+          </div>
+        </td>
+      </tr>
+
+      <tr>
+        <td>BackgroundImage(s)</td>
+        <div
+          v-for="image in profileData?.LSP3Profile?.backgroundImage"
+          :key="image"
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:145]: Same, let's use index as key

File: src/views/profile/ProfileDetail.vue [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        >
+          <pre
+            >{{ image }}
+                </pre
+          >
+          <img
+            :src="image.url.replace('ipfs://', DEFAULT_NETWORK_CONFIG.ipfs.url)"
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:152]: I would make util for this since it's repeated.

File: src/views/profile/ProfileDetail.vue [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    });
+};
+
+const getProfileDataFromIPFS = (ipfsHash: string) => {
+  if (!ipfsHash) {
+    return false;
+  }
+
+  fetch("https://2eff.lukso.dev/ipfs/" + ipfsHash)
+    .then(async (result) => {
```
Comment (dzbo) [inline src/views/profile/ProfileDetail.vue:51]: Also `async`/`await`

File: src/views/profile/__tests__/ProfileDetail.spec.ts [hunk 10]
```diff
@@ -0,0 +1,146 @@
+import ProfileDetail from "@/views/profile/ProfileDetail.vue";
+import { render, waitFor } from "@testing-library/vue";
+import { isAddress } from "ethers/lib/utils";
```
Comment (dzbo) [inline src/views/profile/__tests__/ProfileDetail.spec.ts:3]: Let's use `web3` instead of `ethers`

**Review by dzbo — CHANGES_REQUESTED** [review]

File: src/views/profile/__tests__/ProfileDetail.spec.ts (hunk 10 above)
Comment (andyogaga) [reply src/views/profile/__tests__/ProfileDetail.spec.ts]: I added `isAddress` to the useWeb3 functions

File: src/views/profile/Detail.vue [hunk 11]
```diff
@@ -0,0 +1,170 @@
+<script setup lang="ts">
+import { DEFAULT_NETWORK_CONFIG } from "@/helpers/config";
+import { onMounted, ref, watch } from "vue";
+import useErc725 from "@/compositions/useErc725";
+import { useRoute, RouteLocationNormalizedLoaded } from "vue-router";
+import { createIpfsLink } from "@/utils/createLinks";
+import useWeb3 from "@/compositions/useWeb3";
+
+type LSP3Profile = {
```
Comment (dzbo) [inline src/views/profile/Detail.vue:9]: We have type defined in lsp-factory, let's use it from there

**Review by dzbo — CHANGES_REQUESTED** [review]

File: src/views/profile/Detail.vue (hunk 11 above)
Comment (andyogaga) [reply src/views/profile/Detail.vue:9]: The type produced from there has a problem with version of node, I used it but its having issues

Comment (andyogaga) [reply src/views/profile/Detail.vue:9]: image
Tests fail with ` ReferenceError: TextEncoder is not defined`

Comment (andyogaga) [reply src/views/profile/Detail.vue]: I used a different type after speaking with Callum. I think that solves the issue

**Review by dzbo — APPROVED** [review]
:shipit:

**Pull request merged**